packages/mds-attachment-service/repository/migrations/1603203608358-CreateAttachmentsTable.ts

```diff
@@ -0,0 +1,24 @@
+import { MigrationInterface, QueryRunner } from 'typeorm'
+
+export class CreateAttachmentsTable1603203608358 implements MigrationInterface {
+  name = 'CreateAttachmentsTable1603203608358'
+
+  public async up(queryRunner: QueryRunner): Promise<void> {
+    const [attachments] = await queryRunner.query(
+      `SELECT "table_name" FROM information_schema.tables WHERE "table_catalog" = CURRENT_CATALOG AND "table_schema" = CURRENT_SCHEMA AND "table_name" = 'attachments'`
+    )
+    if (attachments === undefined) {
+      await queryRunner.query(
+        `CREATE TABLE "attachments" ("recorded" bigint NOT NULL DEFAULT (extract(epoch from now()) * 1000)::bigint, "id" bigint GENERATED ALWAYS AS IDENTITY, "attachment_id" uuid NOT NULL, "attachment_filename" character varying(64) NOT NULL, "base_url" character varying(127) NOT NULL, "mimetype" character varying(255) NOT NULL, "thumbnail_filename" character varying(64), "thumbnail_mimetype" character varying(64), CONSTRAINT "attachments_pkey" PRIMARY KEY ("attachment_id"))`
+      )
+      await queryRunner.query(`CREATE INDEX "idx_recorded_attachments" ON "attachments" ("recorded") `)
+      await queryRunner.query(`CREATE UNIQUE INDEX "idx_id_attachments" ON "attachments" ("id") `)
+    }
```

Comment:
I like `IF NOT EXISTS` as well, but we opted to specifically interrogate the schema because a) there are some cases where we need to do something different when the tables exist (as opposed to just doing nothing), and b) certain commands (e.g. `ALTER TABLE ... ADD CONSTRAINT ... FOREIGN KEY` don't support `IF NOT EXISTS`. We opted for a consistent approach. Normally, when all migrations are always applied in a greenfield situation (as opposed to having to be compatible with an existing schema) this would be unnecessary.